server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java

```diff
@@ -168,11 +167,11 @@ private PreparedStatement prepareSelect(Session session) {
     }
 
     @Override
-    public CompletableFuture<Optional<MailDTO>> read(MailRepositoryUrl url, MailKey key) {
-        return executor.executeSingleRow(selectMail.bind()
-            .setString(REPOSITORY_NAME, url.asString())
-            .setString(MAIL_KEY, key.asString()))
-            .thenApply(rowOptional -> rowOptional.map(this::toMail));
+    public Mono<Optional<MailDTO>> read(MailRepositoryUrl url, MailKey key) {
```

Comment:
any reason to not return `Mono<MailDTO>`? I think Mono is also covering the emptyness

Reply:
We have been bitten in the past for trying to change too many things at the same time, in particular when dropping Optional usage.
My strategy here is to change only CompletableFuture to Mono for now and then try to use Mono emptyness.